staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -966,11 +1088,21 @@ func (p *processorListener) pop() {
 func (p *processorListener) run() {
 	// this call blocks until the channel is closed.  When a panic happens during the notification
 	// we will catch it, **the offending item will be skipped!**, and after a short delay (one second)
-	// the next notification will be attempted.  This is usually better than the alternative of never
+	// the next notification will be attempted. This is usually better than the alternative of never
 	// delivering again.
-	stopCh := make(chan struct{})
-	wait.Until(func() {
```

Comment:
The only purpose of `wait.Until` here is to deal with panics. The way it was implemented prevented shutting down the goroutine immediately when the last item in the channel caused a panic because `wait.Until` was stuck in a sleep.

I found that when adding more unit tests and checking for proper termination of goroutines. Insteading of allowing goleak to wait longer, I changed this code so that it sleeps only if really needed (= before processing the next item after a panic).